### PR TITLE
throw exception InvalidArgumentException during validate scheme

### PR DESCRIPTION
**Pull request by ArtsiomBruneuski**
### Fixed Issues (if relevant)

1. magento/magento2#14555

### Manual testing scenarios
A schema class:
class Dummy {
    private $foo;

    public function getFoo() { return $this->foo; }
    public function setFoo($foo) { $this->foo = $foo; }
}
Communication configuration (communication.xml):
<?xml version="1.0"?>
<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:Communication/etc/communication.xsd">
    <topic name="foo_topic" request="Dummy" />
</config>
Run validation for communication configuration. It can be done either automatically if you're using Magento EE, or can be done manually by calling into Magento\Framework\Communication\Config->getTopic("foo_topic")

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=18416) <br/>All committers have signed the CLA.

File: lib/internal/Magento/Framework/Test/Unit/Communication/Config/ValidatorTest.php [hunk 1]
```diff
@@ -0,0 +1,54 @@
+<?php
```
Comment (okorshenko) [inline lib/internal/Magento/Framework/Test/Unit/Communication/Config/ValidatorTest.php:1]: please add copyright to the file

File: lib/internal/Magento/Framework/Test/Unit/Communication/Config/ValidatorTest.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+namespace Magento\Framework\Test\Unit\Communication\Config;
+
+
+use Magento\Framework\Communication\Config\Validator;
+use Magento\Framework\Reflection\MethodsMap;
+use Magento\Framework\Reflection\TypeProcessor;
+
+class ValidatorTest extends \PHPUnit\Framework\TestCase
```
Comment (okorshenko) [inline lib/internal/Magento/Framework/Test/Unit/Communication/Config/ValidatorTest.php:17]: please add annotation to the class the properties

File: lib/internal/Magento/Framework/Test/Unit/Communication/Config/ValidatorTest.php [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+     */
+    public function testValidateResponseSchemaType()
+    {
+        /** @var Validator $validator */
+        $validator = new Validator($this->typeProcessor, $this->methodsMap);
+        $validator->validateResponseSchemaType('123', '123');
+    }
+
+    /**
+     * @expectedException  \LogicException
```
Comment (okorshenko) [inline lib/internal/Magento/Framework/Test/Unit/Communication/Config/ValidatorTest.php:60]: please validate exception message

File: lib/internal/Magento/Framework/Test/Unit/Communication/Config/ValidatorTest.php [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+            ->method('isTypeSimple')
+            ->willReturn(false);
+
+        $this->typeProcessor->expects(static::any())
+            ->method('isTypeSimple')
+            ->willReturn(false);
+    }
+
+    /**
+     * @expectedException  \LogicException
```
Comment (okorshenko) [inline lib/internal/Magento/Framework/Test/Unit/Communication/Config/ValidatorTest.php:48]: please validate exception message

File: lib/internal/Magento/Framework/Reflection/MethodsMap.php [hunk 5]
```diff
@@ -94,6 +94,8 @@ public function getMethodReturnType($typeName, $methodName)
      *  'validatePassword' => 'boolean'
      * ]
      * </pre>
+     * @throws \InvalidArgumentException
```
Comment (okorshenko) [inline lib/internal/Magento/Framework/Reflection/MethodsMap.php]: please add description if which exception will be thrown

File: lib/internal/Magento/Framework/Reflection/MethodsMap.php [hunk 6]
```diff
@@ -148,6 +150,8 @@ public function getMethodParams($serviceClassName, $serviceMethodName)
      *
      * @param string $interfaceName
      * @return array
+     * @throws \ReflectionException
```
Comment (okorshenko) [inline lib/internal/Magento/Framework/Reflection/MethodsMap.php]: please add description if which exception will be thrown

File: lib/internal/Magento/Framework/Communication/Config/Validator.php [hunk 7]
```diff
@@ -109,6 +121,7 @@ public function validateResponseHandlersType($serviceName, $methodName, $handler
      * @param string $typeName
      * @return $this
      * @throws \Exception In case when type is invalid
+     * @throws \InvalidArgumentException
```
Comment (okorshenko) [inline lib/internal/Magento/Framework/Communication/Config/Validator.php]: please add description if which exception will be thrown

**Comment by magento-engcom-team** [comment]
Hi @ArtsiomBruneuski. Thank you for your contribution. 
We will aim to release these changes as part of 2.3.1. 
Please check the release notes for final confirmation.

**Pull request merged**